### PR TITLE
[TASK] Remove CsConvObj and ParserFactory from LanguageService::class (#474)

### PR DESCRIPTION
**Pull request by tomasnorre**
Resolves: #474

### CODE REVIEW

File: src/Rector/v8/v0/RemoveLangCsConvObjAndParserFactoryRector.php [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    /**
+     * @param MethodCall $node
+     */
+    public function refactor(Node $node): ?Node
+    {
+        if (! $this->isMethodStaticCallOrClassMethodObjectType($node, LanguageService::class)) {
+            return null;
+        }
+
+        if ($this->isName($node->name, 'csConvObj')) {
```
Comment (tomasnorre) [inline src/Rector/v8/v0/RemoveLangCsConvObjAndParserFactoryRector.php:39]: I have doing it wrong here. I have changed the function call of `csConvObj()`  and not the property usage.
I'll look into this again.

Comment (sabbelasichon) [reply src/Rector/v8/v0/RemoveLangCsConvObjAndParserFactoryRector.php:39]: Right. And we have to deal with the $GLOBALS['LANG'] here too. So you need the Typo3NodeResolver class and check if we are dealing with a propertyFetch on $GLOBALS. See i.e. https://github.com/sabbelasichon/typo3-rector/blob/master/src/Rector/v9/v5/RefactorPropertiesOfTypoScriptFrontendControllerRector.php#L43

**Comment by tomasnorre** [comment]
This could be an inspiration https://github.com/sabbelasichon/typo3-rector/pull/1685

**Comment by tomasnorre** [comment]
I have the rector working now with both languageService and `$GLOBALS['LANG'], but still fighting with some PHPStan feedback,

File: src/Rector/v8/v0/RemoveLangCsConvObjAndParserFactoryRector.php [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+    {
+        if ($this->isLanguageServiceCall($node)) {
+            return false;
+        }
+
+        if ($node instanceof PropertyFetch) {
+            return true;
+        }
+
+        return ! $this->isMethodStaticCallOrClassMethodObjectType($node, AbstractPlugin::class);
```
Comment (sabbelasichon) [inline src/Rector/v8/v0/RemoveLangCsConvObjAndParserFactoryRector.php:100]: This is unrelated to this Rector. Maybe some relict from the other one.

File: src/Rector/v8/v0/RemoveLangCsConvObjAndParserFactoryRector.php [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+    {
+        if (! (property_exists($node, 'var') && null !== $node->var)) {
+            return false;
+        }
+
+        if ($this->isObjectType($node->var, LanguageService::class)) {
+            return true;
+        }
+
+        if ($this->typo3NodeResolver->isPropertyFetchOnAnyPropertyOfGlobals($node, 'LANG')) {
```
Comment (sabbelasichon) [inline src/Rector/v8/v0/RemoveLangCsConvObjAndParserFactoryRector.php:113]: You can use the constants here too. Typo3NodeResolver::LANG

File: src/Rector/v8/v0/RemoveLangCsConvObjAndParserFactoryRector.php [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+    {
+        if (! (property_exists($node, 'name') && null !== $node->name)) {
+            return null;
+        }
+
+        if (null === $this->getName($node->name)) {
+            return null;
+        }
+
+        if ('csConvObj' === $this->getName($node->name)) {
```
Comment (sabbelasichon) [inline src/Rector/v8/v0/RemoveLangCsConvObjAndParserFactoryRector.php:130]: You could use isName here too for better readibility.

**Review by sabbelasichon — COMMENTED** [review]
Thanks a lot. Besides some minor issues, looks good to me

File: src/Rector/v8/v0/RemoveLangCsConvObjAndParserFactoryRector.php (hunk 3 above)
Comment (tomasnorre) [reply src/Rector/v8/v0/RemoveLangCsConvObjAndParserFactoryRector.php:113]: How would that look like in this case. Cause if doing 

```php
 if (Typo3NodeResolver::LANG) {
            return true;
}
``` 

It's always true and we never this the last return statement

Comment (sabbelasichon) [reply src/Rector/v8/v0/RemoveLangCsConvObjAndParserFactoryRector.php:113]: No, i mean instead of the String LANG here. The call is right.

Comment (tomasnorre) [reply src/Rector/v8/v0/RemoveLangCsConvObjAndParserFactoryRector.php:113]: I see.
